tools/projmgr/src/ProjMgrWorker.cpp

```diff
@@ -1962,21 +1967,45 @@ bool ProjMgrWorker::ExecuteGenerator(std::string& generatorId) {
       return false;
     }
     RteGenerator* generator = generators.at(generatorId);
+
+    // Create generate.yml file with context info and destination
+    string generatorDestination = context.directories.gendir;
+    if (generatorDestination.empty()) {
+      generatorDestination = generator->GetExpandedWorkingDir(context.rteActiveTarget);  // Fallback to working dir if no specific generator directory was specified
+    }
+
+    // Make sure the generatorDestination is absolute
+    if (fs::path(generatorDestination).is_relative()) {
+      generatorDestination = context.rteActiveProject->GetProjectPath() + generatorDestination;
+    }
+
+    // Make the generatorDestination is a folder by adding a '/' to the end
+    if (!generatorDestination.empty() && generatorDestination.back() != '/') {
+      generatorDestination += '/';
+    }
+
+    const auto generatorInputFilePath = ProjMgrYamlEmitter::EmitContextInfo(context, generatorDestination);
+
+    if (!generatorInputFilePath) {
+      ProjMgrLogger::Error("Failed to create the generator input file for '" + generatorId + "'");
+      return false;
+    }
+
+    // Update RteTarget with current generatorInputFilePath that was created
+    context.rteActiveTarget->SetGeneratorInputFilePath(*generatorInputFilePath);
+
     // TODO: review RteGenerator::GetExpandedCommandLine and variables
     //const string generatorCommand = m_kernel->GetCmsisPackRoot() + "/" + generator->GetPackagePath() + generator->GetCommand();
     const string generatorCommand = generator->GetExpandedCommandLine(context.rteActiveTarget);
     if (generatorCommand.empty()) {
       ProjMgrLogger::Error("generator command for '" + generatorId + "' was not found");
       return false;
     }
-    const string generatorWorkingDir = generator->GetExpandedWorkingDir(context.rteActiveTarget);
 
-    // Create generate.yml file with context info and destination
-    ProjMgrYamlEmitter::EmitContextInfo(context, generatorWorkingDir);
 
     error_code ec;
     const auto& workingDir = fs::current_path(ec);
-    fs::current_path(generatorWorkingDir, ec);
+    fs::current_path(generatorDestination, ec);
```

Comment:
fs::current_path(generatorWorkingDir, ec); role was to set working directory for a generator. 
Now generator starts in the the destination directory. Is that expected for all generators?

Reply:
Yes this was intentional on my side at least.

Before my change we used the generatorWorkingDir as current_path, and the `destination` property in the generator input file was aligned with the current_path. So the idea here is the same, since the destination is now a bit more flexible thanks to the `output-dirs[gendir]` user-definable property, the current_path should still follow `destination` which follows `gendir`.

The generator input file is currently listing the project and solution as well, so generators can still find relevant information. 

(We need to make sure this project file connection is kept when moving to the `cbuild.yml`-based format)